src/robotFactory.js

```diff
@@ -1,10 +1,80 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    this.name = name;
+    this.weight = weight;
+    this.chipVersion = chipVersion;
 
-class FlyingRobot {}
+    this.coords = {
+      x: coords.x || 0,
+      y: coords.y || 0,
+    };
+  }
 
-class DeliveryDrone {}
+  goForward(step = 1) {
+    this.coords.y += step;
+  };
+
+  goBack(step = 1) {
+    this.coords.y -= step;
+  };
+
+  goLeft(step = 1) {
+    this.coords.x -= step;
+  };
+
+  goRight(step = 1) {
+    this.coords.x += step;
+  };
+
+  getInfo() {
+    let result = `Robot: ${this.name}, `;
+
+    result += `Chip version: ${this.chipVersion}, Weight: ${this.weight}`;
+
+    return result;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    super(name, weight, coords, chipVersion);
+
+    this.coords.z = coords.z || 0;
+  }
+
+  goDown(step = 1) {
+    this.coords.z -= step;
+  };
+
+  goUp(step = 1) {
+    this.coords.z += step;
+  };
+}
+
+class DeliveryDrone extends FlyingRobot {
+  constructor(name, weight, coords, chipVersion, maxLoadWeight, currentLoad) {
+    super(name, weight, coords, chipVersion);
+    this.maxLoadWeight = maxLoadWeight;
+
+    if (!currentLoad) {
+      this.currentLoad = null;
+    } else {
+      this.currentLoad = currentLoad;
+    }
```

Comment:
You could use a ternary operator here or just set currentLoad as null by default in the constructor
```suggestion
   this.currentLoad = currentLoad ? currentLoad : null;
```

Reply:
Лінтер не видає помилку: 
this.currentLoad = currentLoad ? currentLoad : null;
Unnecessary use of conditional expression for default assignment.